backend/FwHeadless/Services/ProjectSyncStatusService.cs

```diff
@@ -0,0 +1,47 @@
+using System.Collections.Concurrent;
+
+namespace FwHeadless.Services;
+
+public class ProjectSyncStatusService()
+{
+    private ConcurrentDictionary<Guid, ProjectSyncStatus> Status { get; init; } = new();
+
+    public void StartSyncing(Guid projectId)
+    {
+        Status.AddOrUpdate(projectId, (_) => ProjectSyncStatus.Syncing, (_, _) => ProjectSyncStatus.Syncing);
+    }
+
+    public void StopSyncing(Guid projectId)
+    {
+        Status.AddOrUpdate(projectId, (_) => ProjectSyncStatus.ReadyToSync, (_, _) => ProjectSyncStatus.ReadyToSync);
```

Comment:
I'd be more inclined to clear the status (i.e. remove it from the dictionary) when a sync is done, because:
1) Leaving it in the dictionary is only an optimization
2) Leaving it in the dictionary requires a server restart before a project deletion would kick in

Reply:
Done.